PSP Integration candidate: Equuleus-rc1+dev18

dzbaker opened the pull request:
**Checklist (Please check before submitting)**
* [x] I reviewed the [Contributing Guide](https://github.com/nasa/PSP/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
- PR #436. Addresses Issue #435
- PR #439. Addresses Issue #432

**Testing performed**
[cFS Bundle Checks](https://github.com/nasa/cFS/pull/797/checks)
[PSP Checks](https://github.com/nasa/PSP/pull/437/checks)

**Expected behavior changes**
See PRs

**System(s) tested on**
Ubuntu 22.04

**Additional context**
Add any other context about the contribution here.

**Third party code**
If included, identify any third party code and provide text file of license

**Contributor Info - All information REQUIRED for consideration of pull request**
@jphickey

jphickey committed:
Fix #435, add coverage tests for PSP modules

Adds a coverage test implementation for all existing PSP modules.  This
should get 100% coverage on all lines and branches.

jphickey commented:
As there is no rush to get this into the open source build, I suggest postponing this merge until next cycle.  In addition to the RTEMS build issue and the leftover commented-out lines, I found a couple other fixups I'd like to include.  I will add a follow-up commit and we can [re]review next CCB.

jphickey committed:
Fix #432, buffer error in the VxWorks sysmon module

The cpu number to poll was not range checked until _after_ the memset.
This is not the correct order of operations, it must range the element
number in the array before writing/clearing it.

jphickey committed:
Fix #435, correct init in PCS_malloc call

The handler executes after the stub count increments, thus this
needs to check for 1 to indicate the first call, not 0.

dzbaker committed:
Merge pull request #439 from jphickey:fix-432-sysmon

Bump actions/download-artifact from 3 to 4.1.7 in /.github/workflows

dzbaker committed:
Merge pull request #436 from jphickey:fix-435-pspmod-covtests

Fix #435, add coverage tests for PSP modules

dzbaker committed:
Updating documentation and version numbers for equuleus-rc1+dev49

jphickey committed:
HOTFIX #437, correct type in UT time handler

jphickey committed:
HOTFIX #437, Additional type fix for RTEMS 4
